masonry/src/contexts.rs

```diff
@@ -868,6 +878,34 @@ impl LayoutCtx<'_> {
         self.get_child_state_mut(child).needs_layout = false;
     }
 
+    /// Gives the widget a clip path.
+    ///
+    /// A widget's clip path will have two effects:
+    /// - It serves as a mask for painting operations of the widget's children (*not* the widget itself).
+    /// - Pointer events must be inside that path to reach the widget's children.
+    pub fn set_clip_path(&mut self, path: Rect) {
+        trace!("set_clip_path {:?}", path);
+        self.widget_state.clip = Some(path);
+        // TODO - Updating the clip path may have
+        // other knock-on effects we'd need to document.
+        self.widget_state.request_accessibility = true;
+        self.widget_state.needs_accessibility = true;
+        self.widget_state.needs_paint = true;
```

Comment:
I've seen this repeated at least three times. It may make sense to create a function just for marking the state change for needing paint and accessibility.

Reply:
I'd like to keep these kinds of refactor for last, especially if we're only factoring three lines or less. I think there's a benefit to being able to see directly what the code does instead of jumping to another function.